From: Matthew Cengia <redacted>
Date: Tue, 15 Aug 2023 00:38:34 +0000 (+1000)
Subject: fx2lib: Update __interrupt syntax for sdcc>=4.2.3
X-Git-Url: http://sigrok.org/gitweb/?p=sigrok-firmware-fx2lafw.git;a=commitdiff_plain;h=5aab87d358a4585a10ad89277bb88ad139077abd

fx2lib: Update __interrupt syntax for sdcc>=4.2.3

The SDCC user manual https://sdcc.sourceforge.net/doc/sdccman.pdf
section 1.5 "Compatibility with previous versions" on page 11 states:

> In 4.2.3, support for non-parenthesized arguments to __using and
> __interrupt was dropped.

Adjust the imported fx2lib source files to unbreak compilation with
more recent SDCC versions.
---

diff --git a/fx2lib/include/autovector.h b/fx2lib/include/autovector.h
index 495a3126..38f80fcd 100644
--- a/fx2lib/include/autovector.h
+++ b/fx2lib/include/autovector.h
@@ -172,49 +172,49 @@ typedef enum {
 // you must include the predef of these in the file with your main
 // so lets just define them here
 
-void sudav_isr() __interrupt SUDAV_ISR;
-void sof_isr() __interrupt SOF_ISR;
-void sutok_isr() __interrupt SUTOK_ISR;
-void suspend_isr() __interrupt SUSPEND_ISR;
-void usbreset_isr() __interrupt USBRESET_ISR;
-void hispeed_isr() __interrupt HISPEED_ISR;
-void ep0ack_isr() __interrupt EP0ACK_ISR;
-void ep0in_isr() __interrupt EP0IN_ISR;
-void ep0out_isr() __interrupt EP0OUT_ISR;
-void ep1in_isr() __interrupt EP1IN_ISR;
-void ep1out_isr() __interrupt EP1OUT_ISR;
-void ep2_isr() __interrupt EP2_ISR;
-void ep4_isr() __interrupt EP4_ISR;
-void ep6_isr() __interrupt EP6_ISR;
-void ep8_isr() __interrupt EP8_ISR;
-void ibn_isr() __interrupt IBN_ISR;
-void ep0ping_isr() __interrupt EP0PING_ISR;
-void ep1ping_isr() __interrupt EP1PING_ISR;
-void ep2ping_isr() __interrupt EP2PING_ISR;
-void ep4ping_isr() __interrupt EP4PING_ISR;
-void ep6ping_isr() __interrupt EP6PING_ISR;
-void ep8ping_isr() __interrupt EP8PING_ISR;
-void errlimit_isr() __interrupt ERRLIMIT_ISR;
-void ep2isoerr_isr() __interrupt EP2ISOERR_ISR;
-void ep4isoerr_isr() __interrupt EP4ISOERR_ISR;
-void ep6isoerr_isr() __interrupt EP6ISOERR_ISR;
-void ep8isoerr_isr() __interrupt EP8ISOERR_ISR;
-void spare_isr() __interrupt RESERVED_ISR; // not used
+void sudav_isr() __interrupt(SUDAV_ISR);
+void sof_isr() __interrupt(SOF_ISR);
+void sutok_isr() __interrupt(SUTOK_ISR);
+void suspend_isr() __interrupt(SUSPEND_ISR);
+void usbreset_isr() __interrupt(USBRESET_ISR);
+void hispeed_isr() __interrupt(HISPEED_ISR);
+void ep0ack_isr() __interrupt(EP0ACK_ISR);
+void ep0in_isr() __interrupt(EP0IN_ISR);
+void ep0out_isr() __interrupt(EP0OUT_ISR);
+void ep1in_isr() __interrupt(EP1IN_ISR);
+void ep1out_isr() __interrupt(EP1OUT_ISR);
+void ep2_isr() __interrupt(EP2_ISR);
+void ep4_isr() __interrupt(EP4_ISR);
+void ep6_isr() __interrupt(EP6_ISR);
+void ep8_isr() __interrupt(EP8_ISR);
+void ibn_isr() __interrupt(IBN_ISR);
+void ep0ping_isr() __interrupt(EP0PING_ISR);
+void ep1ping_isr() __interrupt(EP1PING_ISR);
+void ep2ping_isr() __interrupt(EP2PING_ISR);
+void ep4ping_isr() __interrupt(EP4PING_ISR);
+void ep6ping_isr() __interrupt(EP6PING_ISR);
+void ep8ping_isr() __interrupt(EP8PING_ISR);
+void errlimit_isr() __interrupt(ERRLIMIT_ISR);
+void ep2isoerr_isr() __interrupt(EP2ISOERR_ISR);
+void ep4isoerr_isr() __interrupt(EP4ISOERR_ISR);
+void ep6isoerr_isr() __interrupt(EP6ISOERR_ISR);
+void ep8isoerr_isr() __interrupt(EP8ISOERR_ISR);
+void spare_isr() __interrupt(RESERVED_ISR); // not used
 // gpif ints
-void ep2pf_isr() __interrupt EP2PF_ISR;
-void ep4pf_isr() __interrupt EP4PF_ISR;
-void ep6pf_isr() __interrupt EP6PF_ISR;
-void ep8pf_isr() __interrupt EP8PF_ISR;
-void ep2ef_isr() __interrupt EP2EF_ISR;
-void ep4ef_isr() __interrupt EP4EF_ISR;
-void ep6ef_isr() __interrupt EP6EF_ISR;
-void ep8ef_isr() __interrupt EP8EF_ISR;
-void ep2ff_isr() __interrupt EP2FF_ISR;
-void ep4ff_isr() __interrupt EP4FF_ISR;
-void ep6ff_isr() __interrupt EP6FF_ISR;
-void ep8ff_isr() __interrupt EP8FF_ISR;
-void gpifdone_isr() __interrupt GPIFDONE_ISR;
-void gpifwf_isr() __interrupt GPIFWF_ISR;
+void ep2pf_isr() __interrupt(EP2PF_ISR);
+void ep4pf_isr() __interrupt(EP4PF_ISR);
+void ep6pf_isr() __interrupt(EP6PF_ISR);
+void ep8pf_isr() __interrupt(EP8PF_ISR);
+void ep2ef_isr() __interrupt(EP2EF_ISR);
+void ep4ef_isr() __interrupt(EP4EF_ISR);
+void ep6ef_isr() __interrupt(EP6EF_ISR);
+void ep8ef_isr() __interrupt(EP8EF_ISR);
+void ep2ff_isr() __interrupt(EP2FF_ISR);
+void ep4ff_isr() __interrupt(EP4FF_ISR);
+void ep6ff_isr() __interrupt(EP6FF_ISR);
+void ep8ff_isr() __interrupt(EP8FF_ISR);
+void gpifdone_isr() __interrupt(GPIFDONE_ISR);
+void gpifwf_isr() __interrupt(GPIFWF_ISR);
 
 #endif
 
diff --git a/fx2lib/lib/interrupts/ep0ack_isr.c b/fx2lib/lib/interrupts/ep0ack_isr.c
index dec4e56a..d216717d 100644
--- a/fx2lib/lib/interrupts/ep0ack_isr.c
+++ b/fx2lib/lib/interrupts/ep0ack_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0ack_isr() __interrupt EP0ACK_ISR {}
+void ep0ack_isr() __interrupt(EP0ACK_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0in_isr.c b/fx2lib/lib/interrupts/ep0in_isr.c
index 2fa99a42..f3d8990e 100644
--- a/fx2lib/lib/interrupts/ep0in_isr.c
+++ b/fx2lib/lib/interrupts/ep0in_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0in_isr() __interrupt EP0IN_ISR {}
+void ep0in_isr() __interrupt(EP0IN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0out_isr.c b/fx2lib/lib/interrupts/ep0out_isr.c
index 043ae70d..a5acedb7 100644
--- a/fx2lib/lib/interrupts/ep0out_isr.c
+++ b/fx2lib/lib/interrupts/ep0out_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0out_isr() __interrupt EP0OUT_ISR {}
+void ep0out_isr() __interrupt(EP0OUT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep0ping_isr.c b/fx2lib/lib/interrupts/ep0ping_isr.c
index 3062d177..7b1ee929 100644
--- a/fx2lib/lib/interrupts/ep0ping_isr.c
+++ b/fx2lib/lib/interrupts/ep0ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep0ping_isr() __interrupt EP0PING_ISR {}
+void ep0ping_isr() __interrupt(EP0PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1in_isr.c b/fx2lib/lib/interrupts/ep1in_isr.c
index d4ecf06f..b666fcb3 100644
--- a/fx2lib/lib/interrupts/ep1in_isr.c
+++ b/fx2lib/lib/interrupts/ep1in_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1in_isr() __interrupt EP1IN_ISR {}
+void ep1in_isr() __interrupt(EP1IN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1out_isr.c b/fx2lib/lib/interrupts/ep1out_isr.c
index 998b7aa8..6c87dc20 100644
--- a/fx2lib/lib/interrupts/ep1out_isr.c
+++ b/fx2lib/lib/interrupts/ep1out_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1out_isr() __interrupt EP1OUT_ISR {}
+void ep1out_isr() __interrupt(EP1OUT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep1ping_isr.c b/fx2lib/lib/interrupts/ep1ping_isr.c
index ff8990e0..c32d437c 100644
--- a/fx2lib/lib/interrupts/ep1ping_isr.c
+++ b/fx2lib/lib/interrupts/ep1ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep1ping_isr() __interrupt EP1PING_ISR {}
+void ep1ping_isr() __interrupt(EP1PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2_isr.c b/fx2lib/lib/interrupts/ep2_isr.c
index 43a31b7d..4de637a3 100644
--- a/fx2lib/lib/interrupts/ep2_isr.c
+++ b/fx2lib/lib/interrupts/ep2_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2_isr() __interrupt EP2_ISR {}
+void ep2_isr() __interrupt(EP2_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2ef_isr.c b/fx2lib/lib/interrupts/ep2ef_isr.c
index e26c5e8a..46f3f87a 100644
--- a/fx2lib/lib/interrupts/ep2ef_isr.c
+++ b/fx2lib/lib/interrupts/ep2ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ef_isr() __interrupt EP2EF_ISR{}
+void ep2ef_isr() __interrupt(EP2EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2ff_isr.c b/fx2lib/lib/interrupts/ep2ff_isr.c
index 22bb95d7..400e4803 100644
--- a/fx2lib/lib/interrupts/ep2ff_isr.c
+++ b/fx2lib/lib/interrupts/ep2ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ff_isr() __interrupt EP2FF_ISR{}
+void ep2ff_isr() __interrupt(EP2FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2isoerr_isr.c b/fx2lib/lib/interrupts/ep2isoerr_isr.c
index b9dc9b50..d0ca4a89 100644
--- a/fx2lib/lib/interrupts/ep2isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep2isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2isoerr_isr() __interrupt EP2ISOERR_ISR {}
+void ep2isoerr_isr() __interrupt(EP2ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep2pf_isr.c b/fx2lib/lib/interrupts/ep2pf_isr.c
index 7cad068e..9bd8e9aa 100644
--- a/fx2lib/lib/interrupts/ep2pf_isr.c
+++ b/fx2lib/lib/interrupts/ep2pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2pf_isr() __interrupt EP2PF_ISR{}
+void ep2pf_isr() __interrupt(EP2PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep2ping_isr.c b/fx2lib/lib/interrupts/ep2ping_isr.c
index 78affed5..dad1e953 100644
--- a/fx2lib/lib/interrupts/ep2ping_isr.c
+++ b/fx2lib/lib/interrupts/ep2ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep2ping_isr() __interrupt EP2PING_ISR {}
+void ep2ping_isr() __interrupt(EP2PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4_isr.c b/fx2lib/lib/interrupts/ep4_isr.c
index 0315c853..5affac9a 100644
--- a/fx2lib/lib/interrupts/ep4_isr.c
+++ b/fx2lib/lib/interrupts/ep4_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4_isr() __interrupt EP4_ISR {}
+void ep4_isr() __interrupt(EP4_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4ef_isr.c b/fx2lib/lib/interrupts/ep4ef_isr.c
index a203e7ba..4a0cff1a 100644
--- a/fx2lib/lib/interrupts/ep4ef_isr.c
+++ b/fx2lib/lib/interrupts/ep4ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ef_isr() __interrupt EP4EF_ISR{}
+void ep4ef_isr() __interrupt(EP4EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4ff_isr.c b/fx2lib/lib/interrupts/ep4ff_isr.c
index e0cd201b..3278da2d 100644
--- a/fx2lib/lib/interrupts/ep4ff_isr.c
+++ b/fx2lib/lib/interrupts/ep4ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ff_isr() __interrupt EP4FF_ISR{}
+void ep4ff_isr() __interrupt(EP4FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4isoerr_isr.c b/fx2lib/lib/interrupts/ep4isoerr_isr.c
index 579fb8bf..dcd7ff90 100644
--- a/fx2lib/lib/interrupts/ep4isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep4isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4isoerr_isr() __interrupt EP4ISOERR_ISR {}
+void ep4isoerr_isr() __interrupt(EP4ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep4pf_isr.c b/fx2lib/lib/interrupts/ep4pf_isr.c
index ec5ce725..7bbe9f02 100644
--- a/fx2lib/lib/interrupts/ep4pf_isr.c
+++ b/fx2lib/lib/interrupts/ep4pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4pf_isr() __interrupt EP4PF_ISR{}
+void ep4pf_isr() __interrupt(EP4PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep4ping_isr.c b/fx2lib/lib/interrupts/ep4ping_isr.c
index 18bac000..8d226856 100644
--- a/fx2lib/lib/interrupts/ep4ping_isr.c
+++ b/fx2lib/lib/interrupts/ep4ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep4ping_isr() __interrupt EP4PING_ISR {}
+void ep4ping_isr() __interrupt(EP4PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6_isr.c b/fx2lib/lib/interrupts/ep6_isr.c
index 552e60de..630f9b26 100644
--- a/fx2lib/lib/interrupts/ep6_isr.c
+++ b/fx2lib/lib/interrupts/ep6_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6_isr() __interrupt EP6_ISR {}
+void ep6_isr() __interrupt(EP6_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6ef_isr.c b/fx2lib/lib/interrupts/ep6ef_isr.c
index 39e8e0c5..e9ba6d54 100644
--- a/fx2lib/lib/interrupts/ep6ef_isr.c
+++ b/fx2lib/lib/interrupts/ep6ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ef_isr() __interrupt EP6EF_ISR{}
+void ep6ef_isr() __interrupt(EP6EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6ff_isr.c b/fx2lib/lib/interrupts/ep6ff_isr.c
index 54137316..011fe871 100644
--- a/fx2lib/lib/interrupts/ep6ff_isr.c
+++ b/fx2lib/lib/interrupts/ep6ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ff_isr() __interrupt EP6FF_ISR{}
+void ep6ff_isr() __interrupt(EP6FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6isoerr_isr.c b/fx2lib/lib/interrupts/ep6isoerr_isr.c
index 407414c7..8c005777 100644
--- a/fx2lib/lib/interrupts/ep6isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep6isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6isoerr_isr() __interrupt EP6ISOERR_ISR {}
+void ep6isoerr_isr() __interrupt(EP6ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep6pf_isr.c b/fx2lib/lib/interrupts/ep6pf_isr.c
index 808f7513..52f6d738 100644
--- a/fx2lib/lib/interrupts/ep6pf_isr.c
+++ b/fx2lib/lib/interrupts/ep6pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6pf_isr() __interrupt EP6PF_ISR{}
+void ep6pf_isr() __interrupt(EP6PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep6ping_isr.c b/fx2lib/lib/interrupts/ep6ping_isr.c
index 53f8ba52..5430f209 100644
--- a/fx2lib/lib/interrupts/ep6ping_isr.c
+++ b/fx2lib/lib/interrupts/ep6ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep6ping_isr() __interrupt EP6PING_ISR {}
+void ep6ping_isr() __interrupt(EP6PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8_isr.c b/fx2lib/lib/interrupts/ep8_isr.c
index 4f1dd51d..d415a3ec 100644
--- a/fx2lib/lib/interrupts/ep8_isr.c
+++ b/fx2lib/lib/interrupts/ep8_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8_isr() __interrupt EP8_ISR {}
+void ep8_isr() __interrupt(EP8_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8ef_isr.c b/fx2lib/lib/interrupts/ep8ef_isr.c
index c018f3ac..d04d2376 100644
--- a/fx2lib/lib/interrupts/ep8ef_isr.c
+++ b/fx2lib/lib/interrupts/ep8ef_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ef_isr() __interrupt EP8EF_ISR{}
+void ep8ef_isr() __interrupt(EP8EF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8ff_isr.c b/fx2lib/lib/interrupts/ep8ff_isr.c
index a6169de3..d1e5fcd7 100644
--- a/fx2lib/lib/interrupts/ep8ff_isr.c
+++ b/fx2lib/lib/interrupts/ep8ff_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ff_isr() __interrupt EP8FF_ISR{}
+void ep8ff_isr() __interrupt(EP8FF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8isoerr_isr.c b/fx2lib/lib/interrupts/ep8isoerr_isr.c
index a95bae9a..a7e8e0f9 100644
--- a/fx2lib/lib/interrupts/ep8isoerr_isr.c
+++ b/fx2lib/lib/interrupts/ep8isoerr_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8isoerr_isr() __interrupt EP8ISOERR_ISR {}
+void ep8isoerr_isr() __interrupt(EP8ISOERR_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ep8pf_isr.c b/fx2lib/lib/interrupts/ep8pf_isr.c
index b0e687c6..53dcdf27 100644
--- a/fx2lib/lib/interrupts/ep8pf_isr.c
+++ b/fx2lib/lib/interrupts/ep8pf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8pf_isr() __interrupt EP8PF_ISR{}
+void ep8pf_isr() __interrupt(EP8PF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/ep8ping_isr.c b/fx2lib/lib/interrupts/ep8ping_isr.c
index e9c25d9e..5cf05dc7 100644
--- a/fx2lib/lib/interrupts/ep8ping_isr.c
+++ b/fx2lib/lib/interrupts/ep8ping_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ep8ping_isr() __interrupt EP8PING_ISR {}
+void ep8ping_isr() __interrupt(EP8PING_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/errlimit_isr.c b/fx2lib/lib/interrupts/errlimit_isr.c
index fe103c22..cf4e1f61 100644
--- a/fx2lib/lib/interrupts/errlimit_isr.c
+++ b/fx2lib/lib/interrupts/errlimit_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void errlimit_isr() __interrupt ERRLIMIT_ISR {}
+void errlimit_isr() __interrupt(ERRLIMIT_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/gpifdone_isr.c b/fx2lib/lib/interrupts/gpifdone_isr.c
index 6bb15c77..ee5ce1f6 100644
--- a/fx2lib/lib/interrupts/gpifdone_isr.c
+++ b/fx2lib/lib/interrupts/gpifdone_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void gpifdone_isr() __interrupt GPIFDONE_ISR{}
+void gpifdone_isr() __interrupt(GPIFDONE_ISR){}
 
diff --git a/fx2lib/lib/interrupts/gpifwf_isr.c b/fx2lib/lib/interrupts/gpifwf_isr.c
index 49eba448..e9941541 100644
--- a/fx2lib/lib/interrupts/gpifwf_isr.c
+++ b/fx2lib/lib/interrupts/gpifwf_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void gpifwf_isr() __interrupt GPIFWF_ISR{}
+void gpifwf_isr() __interrupt(GPIFWF_ISR){}
 
diff --git a/fx2lib/lib/interrupts/hispeed_isr.c b/fx2lib/lib/interrupts/hispeed_isr.c
index 7b05916f..9daabc12 100644
--- a/fx2lib/lib/interrupts/hispeed_isr.c
+++ b/fx2lib/lib/interrupts/hispeed_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void hispeed_isr() __interrupt HISPEED_ISR {}
+void hispeed_isr() __interrupt(HISPEED_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/ibn_isr.c b/fx2lib/lib/interrupts/ibn_isr.c
index 0f38871a..aa0a7523 100644
--- a/fx2lib/lib/interrupts/ibn_isr.c
+++ b/fx2lib/lib/interrupts/ibn_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void ibn_isr() __interrupt IBN_ISR {}
+void ibn_isr() __interrupt(IBN_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sof_isr.c b/fx2lib/lib/interrupts/sof_isr.c
index 49bf5081..64a409c4 100644
--- a/fx2lib/lib/interrupts/sof_isr.c
+++ b/fx2lib/lib/interrupts/sof_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sof_isr() __interrupt SOF_ISR {}
+void sof_isr() __interrupt(SOF_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/spare_isr.c b/fx2lib/lib/interrupts/spare_isr.c
index 965b2818..3b81fd20 100644
--- a/fx2lib/lib/interrupts/spare_isr.c
+++ b/fx2lib/lib/interrupts/spare_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void spare_isr() __interrupt RESERVED_ISR {}
+void spare_isr() __interrupt(RESERVED_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sudav_isr.c b/fx2lib/lib/interrupts/sudav_isr.c
index 7660e115..4d36b59a 100644
--- a/fx2lib/lib/interrupts/sudav_isr.c
+++ b/fx2lib/lib/interrupts/sudav_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sudav_isr() __interrupt SUDAV_ISR {}
+void sudav_isr() __interrupt(SUDAV_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/suspend_isr.c b/fx2lib/lib/interrupts/suspend_isr.c
index 4b2f1aa2..cf08b28a 100644
--- a/fx2lib/lib/interrupts/suspend_isr.c
+++ b/fx2lib/lib/interrupts/suspend_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void suspend_isr() __interrupt SUSPEND_ISR {}
+void suspend_isr() __interrupt(SUSPEND_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/sutok_isr.c b/fx2lib/lib/interrupts/sutok_isr.c
index 05330799..8c7368d2 100644
--- a/fx2lib/lib/interrupts/sutok_isr.c
+++ b/fx2lib/lib/interrupts/sutok_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void sutok_isr() __interrupt SUTOK_ISR {}
+void sutok_isr() __interrupt(SUTOK_ISR) {}
 
diff --git a/fx2lib/lib/interrupts/usbreset_isr.c b/fx2lib/lib/interrupts/usbreset_isr.c
index 7f2ed72d..8888eede 100644
--- a/fx2lib/lib/interrupts/usbreset_isr.c
+++ b/fx2lib/lib/interrupts/usbreset_isr.c
@@ -17,5 +17,5 @@
 
 #include <autovector.h>
 
-void usbreset_isr() __interrupt USBRESET_ISR {}
+void usbreset_isr() __interrupt(USBRESET_ISR) {}
 
